src/invariant/LiabilitiesMatchOffers.cpp

```diff
@@ -96,6 +98,28 @@ checkAuthorized(std::shared_ptr<LedgerEntry const> const& current)
                                    xdr::xdr_to_string(trust));
             }
         }
+
+        if (!isAuthorized(*current))
+        {
+            auto curSellingLiabilities = getSellingLiabilities(*current);
+            auto curBuyingLiabilities = getBuyingLiabilities(*current);
+
+            bool sellingLiabilitiesInc =
+                previous
+                    ? curSellingLiabilities > getSellingLiabilities(*previous)
+                    : curSellingLiabilities > 0;
+            bool buyingLiabilitiesInc =
+                previous
+                    ? curBuyingLiabilities > getBuyingLiabilities(*previous)
+                    : curBuyingLiabilities > 0;
+
+            if (sellingLiabilitiesInc || buyingLiabilitiesInc)
+            {
+                return fmt::format(
+                    "Liabilities increased on unauthorized trust line {}",
```

Comment:
error message is confusing: this can only be reached if `AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG` is set.

The code would be clearer if it was written like this:

```
if (!isAuthorized())
{
     if (isAuthorizedToMaintainLiabilities())
     {
         // check that liabilities can only decrease
     }
     else
     {
          // check that there cannot be any liabilities
     }
}
```

Reply:
done